mypy/nodes.py

```diff
@@ -704,6 +705,28 @@ def deserialize(cls, data: JsonDict) -> 'Decorator':
         return dec
 
 
+class CallableDecorator(FuncItem):
+    """A wrapper around a Decorator that allows it to be treated as a callable function"""
+    def __init__(self, decorator: Decorator) -> None:
+        super().__init__(decorator.func.arguments, decorator.func.body,
+                         cast('mypy.types.CallableType', decorator.type))
```

Comment:
This is better but I think this can be made even simpler. I don't see why this class is necessary, moreover instead of one class that I don't like (`Decorator`), we will have two. Instead of trying to solve everything, why not start from a simple approach that fixes the original issues, and then generalize it to a reasonable extent?

Reply:
The main reason I do it this way is that `FuncBase` and `FuncItem` are both abstract, so I can't instantiate them directly, and I don't know if it makes sense to return a `FuncDef`, since that's not really what this is. I think it makes sense to have a callable decorator be represented as a `FuncBase`, so that code that is looking for something callable can treat it the same as it would any other callable. The alternative is to create a lot of special cases to handle `Decorator`s, which doesn't seem like it serves the long-term interest of eliminating the `Decorator` class altogether.